### PR TITLE
Raise an exception when invalid subshard name

### LINKED ISSUE #112588: CI should fail if invalid subshard name is given.
I think we should fail if invalid subshard name, I spent some time debugging  why my shard was running integration tests and turns out that if you make this mistake  `_selectIndexOfTotalSubshard` returns an empty array which then on `_runDartTest` if you pass an empty array you end up with a command like this `run test --flags` and then it runs every single test on your `workingDirectory`. I will make an issue to fix this

_Originally posted by @Jasguerrero in https://github.com/flutter/flutter/pull/112505#discussion_r982707260_

### PR DESCRIPTION
**Pull request by Jasguerrero**
Fixes: #112588
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: dev/bots/test/test_test.dart [hunk 1]
```diff
@@ -144,5 +144,13 @@ void main() {
       expectExitCode(result, 1);
       expect(result.stdout, contains('Invalid subshard name'));
     });
+
+    test('exits with code 1 when invalid SUBSHARD name', () async {
```
Comment (christopherfujino) [inline dev/bots/test/test_test.dart:148]: code 255, right?

**Review by christopherfujino — APPROVED** [review]
This LGTM. I would only say that be prepared to watch the tree when we land this as there is a good chance some targets will fail. Unless @keyonghan has a better strategy for how to get signal from the devicelab targets without breaking the tree.

**Comment by keyonghan** [comment]
I guess we are well covered by the pre-submit runs where all shard targets enabled in post-submit are also enabled in pre-submit.

**Review by keyonghan — APPROVED** [review]
LGTM

**Pull request merged**